### PR TITLE
[VR-10900] Async get experiment run by id

### PR DESCRIPTION
**Pull request by anandjakhaniya**
**Jira Ticket:** https://vertaai.atlassian.net/browse/VR-10900

### CODE REVIEW

File: backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunServiceImpl.java [hunk 1]
```diff
@@ -60,7 +60,21 @@ public void getExperimentRunsInExperiment(
...
-    super.getExperimentRunById(request, responseObserver);
+    try {
+      final var response =
+          futureExperimentRunDAO
+              .findExperimentRuns(
+                  FindExperimentRuns.newBuilder().addExperimentRunIds(request.getId()).build())
+              .thenApply(
+                  findResponse ->
+                      GetExperimentRunById.Response.newBuilder()
+                          .setExperimentRun(findResponse.getExperimentRuns(0))
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunServiceImpl.java:84]: You have no guarantee you'll have at least one result in the response.

**Review by conradoverta — CHANGES_REQUESTED** [review]

File: backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunServiceImpl.java (hunk 1 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunServiceImpl.java:84]: Throw error if more than one run found for given id

Comment (conradoverta) [reply backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunServiceImpl.java:84]: This is exactly the opposite of what I said. I said that we might not even have one result, as the `get` assumes there is at least one element. I'm not concerned about the case of more than one result.

Comment (conradoverta) [reply backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunServiceImpl.java:84]: We correctly handle the case of none found in https://github.com/VertaAI/modeldb/pull/2239/files

Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunServiceImpl.java:84]: Handled 404 case as well

**Review by conradoverta — APPROVED** [review]

**Pull request merged**